staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,379 @@
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// `items` is a list of request-priorities.
+	Items []PriorityLevelConfiguration `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// PriorityLevelConfigurationSpec is specification of a priority level
+type PriorityLevelConfigurationSpec struct {
+	// `queuingType` indicates whether this priority level does
+	// queuing or is exempt.  Valid values are "Queuing" and "Exempt",
+	// and this field defaults to "Queuing".  "Exempt" means that
+	// requests of this priority level are not subject to concurrency
+	// limits (and thus are never queued) and do not detract from the
+	// concurrency available for non-exempt requests.
+	// +optional
+	QueuingType PriorityLevelQueueingType `json:"queuingType,omitempty" protobuf:"varint,1,opt,name=queuingType"`
+
+	// `queuingConfig` holds the configuration parameters that are
+	// only meaningful for a priority level that does queuing (i.e.,
+	// is not exempt).  This field must be non-empty if and only if
+	// `queuingType` is `"Queuing"`.
+	// +optional
+	QueuingConfig *QueuingConfiguration `json:"queuingConfig,omitempty" protobuf:"bytes,2,opt,name=queuingConfig"`
+}
+
+// PriorityLevelQueueingType identifies the queuing nature of a priority level
+type PriorityLevelQueueingType string
+
+// PriorityLevelQueuingTypeQueueing is the PriorityLevelQueueingType for priority levels that queue
+const PriorityLevelQueuingTypeQueueing PriorityLevelQueueingType = "Queuing"
+
+// PriorityLevelQueuingTypeExempt is the PriorityLevelQueueingType for priority levels that are exempt from concurrency controls
+const PriorityLevelQueuingTypeExempt PriorityLevelQueueingType = "Exempt"
+
+// QueuingConfiguration holds the configuration parameters that are specific to a priority level that is subject to concurrency controls
+type QueuingConfiguration struct {
+	// `assuredConcurrencyShares` (ACS) must be a positive number. The
+	// API server's concurrency limit (SCL) is divided among the
+	// concurrency-controlled priority levels in proportion to their
+	// assured concurrency shares. This produces the assured
+	// concurrency value (ACV) for each such priority level:
+	//
+	//             ACV(l) = ceil( SCL * ACS(l) / ( sum[priority levels k] ACS(k) ) )
+	//
+	// Required.
+	AssuredConcurrencyShares int32 `json:"assuredConcurrencyShares" protobuf:"varint,1,opt,name=assuredConcurrencyShares"`
+
+	// `queues` is the number of queues for this priority level. The
+	// queues exist independently at each apiserver. The value must be
+	// positive.  Setting it to 1 effectively precludes
+	// shufflesharding and thus makes the distinguisher method of
+	// associated flow schemas irrelevant.  This field has a default
+	// value of 64.
+	// +optional
+	Queues int32 `json:"queues" protobuf:"varint,2,opt,name=queues"`
+
+	// `handSize` is a small positive number that configures the
+	// shuffle sharding of requests into queues.  When a request
```

Comment:
Sorry, I realized this sentence skips over the multiplicity of priority levels.  The phrase "When a request arrives at an apiserver" should be replaced with something like "When enqueuing a request at this priority level".

Reply:
fixed